lib/fuzzy.dart

```diff
@@ -30,6 +30,8 @@ class Fuzzy<T> {
 
   /// Search for a given [pattern] on the [list], optionally [limit]ing the result length
   List<Result<T>> search(String pattern, [int limit = -1]) {
+    if (list.isEmpty) return <Result<T>>[];
```

Comment:
Given this bug, I think `list` can be sometimes `null`.  Could you create a test to guarantee it won't be? (and maybe set it to a `const []` on constructor if `null` is received?